### ISSUE TITLE
ignore_errors doesn't update failed task counter in a Play Recap

### ISSUE THREAD

**Issue by alexxa** [ansible/ansible#27160]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

 - ignore_errors
 - failed_when
##### ANSIBLE VERSION

Tested on `ansible-2.3.1` and
```
ansible-2.4.0-100.git201707210442.a53dabf.devel.fc24.noarch

```

##### SUMMARY

1) An error message is not shown with `register` + `failed_when`, if a task fails. IMHO, it should be shown as `ignore_errors: yes` does. Compare output of Task_3 and Task_4 below. Taks_4 fails, but I don't see it in a Task output or in a Play Recap. Both tasks have status `changed`. How will I know that the error occurred? 
2) There is no sign of occurred errors/fails in a Play Recap. `failed=0` is shown, however two tasks failed. How will I know that the error occurred? Should I scroll over all playbook output to look for `...ignoring` or `fatal...` outputs?  Failed increment should be updated accordingly, or an extra increment, e.g. `ignored_failers = counter` should be added.

##### STEPS TO REPRODUCE

```
# This is a playbook to demonstrate error handling and a failed increment
- name: using error handlers
  hosts: localhost
  connection: local
  tasks:

    - name: Task_1, ls existing file, no error expected
      raw:  ls file.txt
    - name: Task_2, ls non existing file, ignore error
      raw: ls no_file.txt
      ignore_errors: yes
    - name: Task_3, ls existing file, register an error, failed_when
      raw: ls file.txt
      register: command_result
      failed_when: "'FAILED' in command_result.stderr"
    - name: Task_4, ls not existing file, register an error
      raw: ls no_file.txt
      register: command_result
      failed_when: "'FAILED' in command_result.stderr"

```

##### ACTUAL RESULTS

```
PLAY [using error handlers] ***************************************************************************************

TASK [Gathering Facts] ********************************************************************************************

TASK [Task_1, ls existing file, no error expected] **************************************************************
changed: [127.0.0.1]

TASK [Task_2, ls non existing file, ignore error] ****************************************************************
fatal: [127.0.0.1]: FAILED! => {"changed": true, "failed": true, "msg": "non-zero return code", "rc": 2, "stderr": "ls: cannot access 'no_file.txt': No such file or directory\n", "stdout": "", "stdout_lines": []}
...ignoring

TASK [Task_3, ls existing file, register an error, failed_when] ************************************************
changed: [127.0.0.1]

TASK [Task_4, ls not existing file, register an error] ************************************************************
changed: [127.0.0.1]

PLAY RECAP *********************************************************************************************************
127.0.0.1                  : ok=5    changed=4    unreachable=0    failed=0 
<!--- Paste verbatim command output between quotes below -->
```

```

**Comment by bcoca** [comment]
1. not a bug, `failed_when` decides if a task is failed or not, it sets the status so that task is NOT failed.

2. bug, ignore_errors should still count as failed task ... or at least an 'ignored' counter as you mention.

**Comment by guzzijason** [comment]
Re: item 2, with ansible-2.7.0, I still see `failed=0` in the play recap if `ignore_errors: yes`. Desired behavior would be to count failures, but not abort the play because of them.

**Comment by danielbjorkholm** [comment]
I agree with @guzzijason 
Using some ansible playbooks for pure validation/testing, I would love to run all tasks in `ignore_errors: yes` and still get a burning red counter in the end of what would then be a very neat-looking test report.

**Comment by bcoca** [comment]
fixed via be9f07279e3 (now shows as ignored)
